File: app/src/main/java/com/sarris/poker/card/CardPresenterModule.java
package com.sarris.poker.card;

import dagger.Module;
import dagger.Provides;

/**
 * Created by georgiossarris on 5/24/17.
 */

@Module
public class CardPresenterModule {

    private final CardContract.View cardView;

    CardPresenterModule(CardContract.View view) {
        cardView = view;
    }

    @Provides
    CardContract.View providesCardContractView() {
        return cardView;
    }

}

File: app/src/main/java/com/sarris/poker/card/CardPresenterModule.kt
package com.sarris.poker.card

import dagger.Module
import dagger.Provides

/**
 * Created by georgiossarris on 5/24/17.
 */
@Module
class CardPresenterModule internal constructor(private val cardView: CardContract.View) {
    @Provides
    fun providesCardContractView(): CardContract.View {
        return cardView
    }
}
